Add "core-js" as dependency to @apollo/federation

Upgrading from 0.23.0 to 0.23.1 broke my build with

```
 Error: Cannot find module 'core-js/features/array/flat'
  Require stack:
  - /home/starptech/repositories/starptech/demo/node_modules/@apollo/federation/dist/index.js
```

0.23.0

```
core-js@3.11.0
node_modules/core-js
  core-js@"^3.0.1" from apollo-env@0.6.6
  node_modules/apollo-env
    apollo-env@"^0.6.6" from apollo-graphql@0.6.1
    node_modules/apollo-graphql
      apollo-graphql@"^0.6.0" from @apollo/federation@0.23.0
      node_modules/@apollo/federation
        @apollo/federation@"0.23.0" from the root project
```

0.23.1

```
❯ npm why core-js
npm ERR! No dependencies found matching core-js
```

--- a/federation-integration-testsuite-js/package.json
+++ b/federation-integration-testsuite-js/package.json
@@ -21,7 +21,7 @@
     "node": ">=12"
   },
   "dependencies": {
-    "apollo-graphql": "^0.9.0",
+    "apollo-graphql": "^0.9.2",
     "graphql-tag": "^2.10.4"
   }
 }

--- a/federation-js/CHANGELOG.md
+++ b/federation-js/CHANGELOG.md
@@ -4,7 +4,7 @@
 
 > The changes noted within this `vNEXT` section have not been released yet.  New PRs and commits which introduce changes should include an entry in this `vNEXT` section as part of their development.  When a release is being prepared, a new header will be (manually) created below and the appropriate changes within that release will be moved into the new section.
 
-- Remove lingering `core-js` polyfill imports, they're no longer needed and their presence is problematic since `core-js` isn't defined as a dependency within the package. Update `apollo-graphql` dependency which resolves a missing dependency (`sha.js`) within that package. [PR #699](https://github.com/apollographql/federation/pull/699)
+- Remove lingering `core-js` polyfill imports, they're no longer needed (since `@apollo/gateway@0.15.0` dropped support for <= Node.js v10) and their presence is problematic since `core-js` isn't defined as a dependency within the package. Update `apollo-graphql` dependency which resolves a missing dependency (`sha.js`) within that package. [PR #699](https://github.com/apollographql/federation/pull/699)
 
 ## v0.23.1
 

--- a/federation-js/CHANGELOG.md
+++ b/federation-js/CHANGELOG.md
@@ -4,7 +4,7 @@
 
 > The changes noted within this `vNEXT` section have not been released yet.  New PRs and commits which introduce changes should include an entry in this `vNEXT` section as part of their development.  When a release is being prepared, a new header will be (manually) created below and the appropriate changes within that release will be moved into the new section.
 
--  _Nothing yet! Stay tuned!_
+- Remove lingering `core-js` polyfill imports, they're no longer needed and their presence is problematic since `core-js` isn't defined as a dependency within the package. Update `apollo-graphql` dependency which resolves a missing dependency (`sha.js`) within that package. [PR #699](https://github.com/apollographql/federation/pull/699)
 
 ## v0.23.1
 

--- a/federation-js/package.json
+++ b/federation-js/package.json
@@ -23,7 +23,7 @@
     "access": "public"
   },
   "dependencies": {
-    "apollo-graphql": "^0.9.0",
+    "apollo-graphql": "^0.9.2",
     "lodash.xorby": "^4.7.0"
   },
   "peerDependencies": {

--- a/federation-js/src/index.ts
+++ b/federation-js/src/index.ts
@@ -1,5 +1,2 @@
-import "core-js/features/array/flat";
-import "core-js/features/array/flat-map";
-
 export * from './composition';
 export * from './service';

--- a/gateway-js/CHANGELOG.md
+++ b/gateway-js/CHANGELOG.md
@@ -4,7 +4,7 @@
 
 > The changes noted within this `vNEXT` section have not been released yet.  New PRs and commits which introduce changes should include an entry in this `vNEXT` section as part of their development.  When a release is being prepared, a new header will be (manually) created below and the appropriate changes within that release will be moved into the new section.
 
--  _Nothing yet! Stay tuned!_
+- Update `apollo-graphql` dependency which resolves a missing dependency (`sha.js`) within that package. [PR #699](https://github.com/apollographql/federation/pull/699)
 
 ## v0.26.2
 

--- a/gateway-js/package.json
+++ b/gateway-js/package.json
@@ -28,7 +28,7 @@
     "@apollo/federation": "file:../federation-js",
     "@apollo/query-planner": "file:../query-planner-js",
     "@types/node-fetch": "2.5.10",
-    "apollo-graphql": "^0.9.0",
+    "apollo-graphql": "^0.9.2",
     "apollo-reporting-protobuf": "^0.6.0",
     "apollo-server-caching": "^0.6.0",
     "apollo-server-core": "^2.23.0",

